src/alire/alire-directories.adb

```diff
@@ -12,10 +12,64 @@ with Alire.TTY;
 
 with GNATCOLL.VFS;
 
+with SI_Units.Binary;
+
 package body Alire.Directories is
 
    package Adirs renames Ada.Directories;
 
+   -------------------
+   -- Temp_Registry --
+   -------------------
+   --  To be able to remove temp files when we are forcibly interrupted, we
+   --  keep track of them here. Calling Delete_Temporaries will do the cleanup
+   --  (as file ops are blocking and cannot be done in a protected).
+   protected Temp_Registry is
+
+      procedure Add (Path : Any_Path);
+      --  Add a path to a temporary
+
+      procedure Del (Path : Any_Path);
+      --  Remove a path to a temporary
+
+      function Get return Utils.String_Set;
+      --  Retrieve all current temporaries
+
+   private
+
+      Registry : Utils.String_Set;
+
+   end Temp_Registry;
+
+   protected body Temp_Registry is
+
+      ---------
+      -- Add --
+      ---------
+
+      procedure Add (Path : Any_Path) is
+      begin
+         --  Store absolute, so CWD changes do not affect us
+         Registry.Include (Ada.Directories.Full_Name (Path));
+      end Add;
+
+      ---------
+      -- Del --
+      ---------
+
+      procedure Del (Path : Any_Path) is
+      begin
+         Registry.Exclude (Path);
```

Comment:
Will the relative path match the absolute path added above?

I think all the subprograms here should probably only accept absolute paths.

Reply:
You're right. I think these paths are all absolute anyway or it wouldn't be working already, but I'll make sure.

Reply:
Should be fixed now.